app.go

```diff
@@ -426,7 +426,22 @@ func dumpCollectionTo(connStr, database, collection string, writer io.Writer, mo
 	session.SetPrefetch(1.0)
 	defer session.Close()
 
-	q := session.DB(database).C(collection).Find(nil).Snapshot()
+	q := session.DB(database).C(collection).Find(nil)
+
+	// https://docs.mongodb.com/manual/release-notes/3.6-compatibility/#deprecated-options
+	var status bson.M
+	err = session.Run(bson.D{{Name: "serverStatus", Value: 1}}, &status)
+	if err != nil {
+		return err
+	}
+
+	storageEngineConf := status["storageEngine"].(bson.M)
+	switch storageEngineConf["name"] {
+	case "mmapv1":
+		q = q.Hint("_id")
```

Comment:
What is this doing and why?

Reply:
This will return documents in the order they were inserted. With `mmapv1` storage engine using natural order won't give us the assurance that every document is returned once, because updates can trigger document relocation.

Reply:
Is this the same approach as how the mongodb dump tools work?

Reply:
I don't think so. mongodump does not provide a consistent backup, unless it is configured to use the operation log to keep data consistent

Reply:
@mjgarton Are you ok with me merging this?